src/BrandwatchReactAuth.js

```diff
@@ -0,0 +1,61 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import jwtDecode from 'jwt-decode';
+import TokenStore from 'viziaauth';
+
+export default class BrandwatchReactAuth extends Component {
+  getChildContext() {
+    return {
+      brandwatchAuthLogout: this.handleLogout,
+    };
+  }
+
+  constructor(props) {
+    super(props);
+    this.store = new TokenStore(this.props.domain),
+    this.handleLogout = this.handleLogout.bind(this);
+
+    this.state = {
+      loggedIn: undefined,
+    };
+
+    props.onCreateStore(this.store);
+  }
+
+  componentWillMount() {
+    this.store.getToken({ aud: this.props.audience }).then((token) => {
+      if (!token) {
+        return window.location.href = this.store.loginUrl;
```

Comment:
I was googling the difference between 

`window.location.href = url` and `window.location = url`

If I'm reading this right it means with the current approach the user can click back, but they will then just got forwarded on again due to a lack of token. (apologies if this isn't the case, I haven't checked it out yet)

https://stackoverflow.com/questions/1865837/whats-the-difference-between-window-location-and-window-location-replace